usecases/linearFunction/test_optimize.py

```diff
@@ -51,26 +51,52 @@ def __init__(self, yaml_file_metadata, yaml_file_data):
 
 class TestOptimize(unittest.TestCase):
     def test_linear_virtual_data(self):
+        virtual_experiment_metadata_file = \
+            Path(__file__).parent / 'virtual_experiment_linear_model_meta.yaml'
+        with open(virtual_experiment_metadata_file, "r") as f:
+            d = yaml.load(f, Loader=yaml.FullLoader)
+
         all_experiments_model_error = \
             MultiLinearModelError('virtual_experiment_linear_model_meta.yaml',
                                   'virtual_experiment_linear_model_data.yaml')
 
         start_vector = np.array([0.7])
         result = least_squares(all_experiments_model_error, start_vector)
 
+        # assert quadratic component of virtual experiment to be zero
+        self.assertAlmostEqual(d['c'], 0.)
+        # check to obtain exact linear coefficient
+        self.assertAlmostEqual(result.x[0], d['b'], 7)
+        # check cost function to be zero
+        self.assertAlmostEqual(result.cost, 0)
+
+    def test_linear_virtual_data_split(self):
         virtual_experiment_metadata_file = \
-            Path(__file__).parent / 'virtual_experiment_linear_model_meta.yaml'
+            Path(__file__).parent / 'virtual_experiment_linear_model_with_noise_meta.yaml'
         with open(virtual_experiment_metadata_file, "r") as f:
             d = yaml.load(f, Loader=yaml.FullLoader)
 
+        all_experiments_model_error = \
+            MultiLinearModelError('virtual_experiment_linear_model_with_noise_meta.yaml',
+                                  'virtual_experiment_linear_model_with_noise_data.yaml')
+
+        def weighted_all_experiments_model_error(prm_vector):
+            all_model_errors = all_experiments_model_error.evaluate(prm_vector)
+            obj = 0.
+            for key, me in all_model_errors.items():
+                obj = obj + np.dot(me['f'], me['f']) / d['sigma_noise_function'] ** 2 + \
```

Comment:
That looks like a likelhoodfunction and _could_ be separate method in `bayes` at some point.